README.md

```diff
@@ -2,4 +2,481 @@
+
+* `stringIsPalindrome:` which takes one `NSString` argument called `string` and returns a `BOOL`, and
+* `stringByReversingString:` which takes one `NSString` argument called `string` and returns an `NSString`.
+
+After declaring these methods, your `FISAppDelegate.h` header file should look something like this:
+
+```objc
+//  FISAppDelegate.h
+
+#import <UIKit/UIKit.h>
+
+@interface FISAppDelegate : UIResponder <UIApplicationDelegate>
+
+@property (strong, nonatomic) UIWindow *window;
+
+- (BOOL)stringIsPalindrome:(NSString *)string;
+
+- (NSString *)stringByReversingString:(NSString *)string;
+
+@end
+```
+Great!
+
+### II. Define Method Implementations
+
+Now, navigate to the `FISAppDelegate.m` implementation file. *Below* the ending curly brace of the `application:didFinishLaunchingWithOptions:` method, but *above* the `@end` statement, use autocomplete to define both methods with default returns:
+
+* `stringIsPalindrome:` should `return NO;`, and
+* `stringByReversingString:` should `return nil;`.
+
+Your `FISAppDelegate.m` file should now look something like this:
+
+```objc
+//  FISAppDelegate.m
+
+#import "FISAppDelegate.h"
+
+@implementation FISAppDelegate
+
+- (BOOL)application:(UIApplication *)application didFinishLaunchingWithOptions:(NSDictionary *)launchOptions {
+    // Override point for customization after application launch.
+    
+    /**
+     
+     * Write your check code here.
+     
+     */
+
+    // do not alter
+    return YES;  //
+}   ///////////////
+
+/**
+ 
+ * Implement your methods here.
+ 
+ */
+
+- (BOOL)stringIsPalindrome:(NSString *)string {
+   return NO;
+}
+
+- (NSString *)stringByReversingString:(NSString *)string {
+    return nil;
+}
+
+@end
+```
+
+### III. Reversing a String
+
+Because the logic of our `stringIsPalindrome:` method depends on our ability to reverse a string, let's start by writing out the `stringByReversingString:` method that we declared. Our goal is take whatever string variable is passed into the `string` argument (also "parameter") and return its exact reverse at the end of the method.
+
+How would we approach this goal with a word or sentence written on paper? Couldn't we simply start at the end of the word or sentence, moving backwards while copying each letter onto a new line? Of course we could!
+
+How could we represent this process in code? Well, we already know that strings are made up of individual characters that are arranged by index. So, could we simply start with the character at the string's highest index and work backwards, copying each character into a new string? Of course we could!
+
+**Advanced:** *There are more efficient ways to reverse a string, but this is perhaps the simplest one to understand.*
+
+One possible implementation of this process in Objective-C is written below:
+
+```objc
+- (NSString *)stringByReversingString:(NSString *)string {
+
+    NSString *result = @"";
+    
+    for (NSUInteger i = [string length]; i > 0; i--) {
```

Comment:
oh that's right, `i >= 0` generates a warning "comparison of unsigned expression to >= 0 is always true"